tests/test_docker_compose.py

```diff
@@ -23,6 +25,22 @@ def test_can_pull_images_before_spawning_service_via_compose():
         assert port == "4444"
 
 
+def test_can_build_images_before_spawning_service_via_compose():
+    with patch(
+        f"{subprocess.call.__module__}.{subprocess.call.__name__}",
+        side_effect=subprocess.call
+    ) as call_mock:
+        with DockerCompose("tests", build=True) as compose:
+            host = compose.get_service_host("hub", 4444)
+            port = compose.get_service_port("hub", 4444)
+            assert host == "0.0.0.0"
+            assert port == "4444"
+
+    assert compose.build
+    docker_compose_cmd = call_mock.call_args_list[0][0][0]
+    assert "docker-compose" in docker_compose_cmd and "--build" in docker_compose_cmd
```

Comment:
I think this test can be simplified
```python
def test_can_build_images_before_spawning_service_via_compose():
    with patch.object(DockerCompose, "_run_command") as call_mock:
        with DockerCompose("tests", build=True):
            assert "--build" in call_mock.call_args_list[0].args[0]
```

It only needs to move the `subprocess.call` to a `DockerCompose` method and replace all `subprocess` calls in class:
```python
class DockerCompose(object):
...
    def _run_command(self, cmd):
        subprocess.call(cmd, cwd=self.filepath)
```

Reply:
This test is weaker as it does not check that `build=True` still runs docker-compose up.  I at least add a check `docker-compose` and `up` are called in the docker-compose subprocess.  Look ok?

Reply:
Thank you. Just two small questions left (see below)